lib/shopify-cli/forms/deploy_script.rb

```diff
@@ -0,0 +1,29 @@
+require 'shopify_cli'
+require 'uri'
+
+module ShopifyCli
+  module Forms
+    class DeployScript < Form
+      flag_arguments :api_key
+
+      def ask
+        self.api_key ||= ask_api_key
+      end
+
+      private
+
+      def ask_api_key
+        apps = Helpers::Organizations.fetch_apps(ctx)
```

Comment:
Why are we asking for which app? Shouldnt this have been part of the create process? This should probably follow the exact same lines how we create every other project and persist an shopify-cli.yml file so that this is not asked every time. Why did you not try to re-use the app creation flow and extend that, so that we could just create an api client for the developer? I assume we would need to change the mutation to allow for your configuration but this is a terrible experience is it not?

Reply:
(from my understanding of your question) By design, a script is not paired 1:1 with an app. It is created separately, and then the same script definition can be deployed to many apps.  The same app script can then be used on multiple shops.  The only restriction is that the shop has to have that app installed